### PR TITLE
Ashvds/doco udate fixes #55

### LINKED ISSUE #55: Step by Step tutorial for getting up and running with hx-deploy-tool
Github pages or wiki would be fine, anywhere really as it'll be better than what we have now (ie, nothing).

### PR DESCRIPTION
**Pull request by ashvds**
Creates the first draft of a user guide.
Also fixes help text which wasn't updated when `select` was replaced with `start`.

Requires the github site to point to the userguide file.

### CODE REVIEW

File: docs/user-guide.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+## What is hx-deploy-tool
+
+A small yet versatile tool to automate the annoying bits of deployment, making it more likely engineers do deployments right(tm) and make deployments a joy.
+Can work with any deployment scripts ('bare metal') and designed to work well with docker in aws.
+It pulls config and release zip files from various sources (eg. local or S3), unpacks into versioned directories, and runs commands to start, stop, and expose them as required.
+
+## Why use hx-deploy-tool
+
+You've decided that you want finer control over your environment than running it in an aplication or function PAAS, and you need to manage your deployments.
```
Comment (timbod7) [inline docs/user-guide.md:17]: typo

File: docs/user-guide.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+You have a single application or service, and when deploying, need to orcestrate various actions, such as deploying the containers to a single or multiple servers, setting up reverse proxies, managing associating certificates, making sure all the containers are the same version of the application, making sure the logs get ingested etc. etc.
+
+hx-deploy-tool provides a reliable way to deploy all those pieces, and ensure state and version is managed at the same time. It encapsulates all the steps of a deployment, without the need to write script after script. Simply create a docker-compose file, and a couple of moustache templates, and suddently with hx-deploy tool, your deployments are consistent, repeatable, reproducable and highly configureable. On top of that, you can run the tool in slave mode, and it will monitor a blob store for the latest state, and ensure the server it is running on, is in that state (release version, configuration etc.)
```
Comment (timbod7) [inline docs/user-guide.md:20]: typo: suddently

File: docs/user-guide.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+There is a gap between running software on a single server, and having large container clusters and using various tools for container orhestration. This tool plugs that gap, and can provide consistency and reliability in deployments until you have grown to a place where you need what container orchestration tools provide.
```
Comment (timbod7) [inline docs/user-guide.md:22]: typo: orhestration

File: docs/userguide/minimum-requirements.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# Minimum Requirements
+
+## Minimum Requirements - no proxy sidecar
+
+At a minimum, you will need the following (with example files included in the release):
```
Comment (timbod7) [inline docs/userguide/minimum-requirements.md:5]: I think it's confusing that you list optional requirements in the "minimum requirements section".

File: docs/userguide/minimum-requirements.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+- Server where you will be deploying your release
+- Configuration file (examples included in the release)
+- Release archive - A versioned .zip file Stored in S3 containing:
```
Comment (timbod7) [inline docs/userguide/minimum-requirements.md:9]: It doesn't need to be S3. I'd link to details of the contents of the zip file, and  write something something like:

- Release store - keeps all published [release zip archives](... link to details of the zip structure...). The release store may either be a path in the local file system or a path in an s3 bucket.

File: docs/userguide/minimum-requirements.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  - release.json - (examples included in the release) - a config for deployment including predeploy, deploy and stop commands to execute. Also declares template files.
+  - **_Optional_** Template files - mustache template files for configuration of other components such as proxy, fluentd and docker compose (based on deploy configuration).
+  - **_Optional_** Deployment Scripts
+  - **_Optional_** Binary to be deployed (the deploy command/script can source the binary)
+- **_Optional_** AWS Credentials: see the [official AWS credential page](https://aws.amazon.com/blogs/security/a-new-and-standardized-way-to-manage-credentials-in-the-aws-sdks/)
+- S3 bucket(s) with necessary permissions for r/w - release store and optional state store
```
Comment (timbod7) [inline docs/userguide/minimum-requirements.md:15]: As per the above, S3 is not necessarily required.

File: docs/userguide/minimum-requirements.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+- Folders with necessary permissions for writing logs
```
Comment (timbod7) [inline docs/userguide/minimum-requirements.md:16]: I'd write something like:

- The following working directories, writable by the user(s) who will run hx-deploy-tool.

```
/opt/releases     # where releases get unpacked for deployment
/opt/etc/deployment  # where deployment contexts get written
/opt/var/log/    # where the log file gets written
```

Note that the actual directories used can be overwritten in the configuration file.

File: docs/userguide/proxy-example.md [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+## 1.1. Download the latest release of hx-deploy-tool
+
+Seems a little obvious, but the release includes a bunch of goodies that you will need to make the deployment work.
+You can find the minimum requirements [here](https://helix-collective.github.io/hx-deploy-tool/userguide/minimum-requirements), so have a read before you try to get things running.
+
+We are going to use hx-deploy-tool to deploy a release on a single machine, with an automatically configured nginx reverse proxy.
+
+## 1.2. Prepare your environment
+
+In the latest release package, you will find several usefull files to help you get started:
```
Comment (timbod7) [inline docs/userguide/proxy-example.md:14]: typo

**Comment by timbod7** [comment]
As discussed today we are going to iterate on this stuff. But it's a great start, so lets commit and push forward.

**Review by timbod7 — APPROVED** [review]

File: docs/user-guide.md (hunk 1 above)
Comment (ashvds) [reply docs/user-guide.md:17]: fixed

File: docs/user-guide.md (hunk 2 above)
Comment (ashvds) [reply docs/user-guide.md:20]: fixed

File: docs/user-guide.md (hunk 3 above)
Comment (ashvds) [reply docs/user-guide.md:22]: fixed

File: docs/userguide/minimum-requirements.md (hunk 5 above)
Comment (ashvds) [reply docs/userguide/minimum-requirements.md:9]: updated with your verbage

File: docs/userguide/minimum-requirements.md (hunk 6 above)
Comment (ashvds) [reply docs/userguide/minimum-requirements.md:15]: added `or directory`

File: docs/userguide/minimum-requirements.md (hunk 7 above)
Comment (ashvds) [reply docs/userguide/minimum-requirements.md:16]: added with minor changes

File: docs/userguide/minimum-requirements.md (hunk 4 above)
Comment (ashvds) [reply docs/userguide/minimum-requirements.md:5]: removed optional lines

File: docs/userguide/proxy-example.md (hunk 8 above)
Comment (ashvds) [reply docs/userguide/proxy-example.md:14]: useful catch

**Comment by ashvds** [comment]
Addressed all comments in this commit https://github.com/helix-collective/hx-deploy-tool/commit/8a14ffc59e8e7c8bfdfb3bf05bfde3ce00e6f07c
Moving the updates to gh-pages branch as discussed.
Closing the pull request, which will be re-created from the other branch when ready

**Pull request closed without merging**